docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -0,0 +1,1382 @@
+
+[source,bash]
+----
+docker build -t fedora-tools:v1 .
+----
+
+Then, run the docker container as:
+
+[source,bash]
+----
+$ docker run --privileged \
+-t -i -v $(PWD)/$(project):/data --rm -p 8080:8080 fedora-tools:v1 /bin/bash
+...
+[root@75d1df96849c data]# _
+----
+
+[[NOTE]]
+====
+Note that in order to use `perf` to profile the native binaries created in the training,
+the container needs to run as privileged, or with `--cap-add sys_admin`.
+Please note that privileged containers are **NOT** recommended in production, so use this flag with caution!
+====
+
+Once the container is running, you need to ensure that the kernel is ready for the profiling exercises:
+
+[source,bash]
+----
+[root@75d1df96849c data]# echo -1 | sudo tee /proc/sys/kernel/perf_event_paranoid
+[root@75d1df96849c data]# echo 0 | sudo tee /proc/sys/kernel/kptr_restrict
+----
+
+Once you’re inside the container, you can execute `strings`, `perf`, `objdump`…, etc commands on the generated binary.
+Since the binary was created inside a Linux container, the container tools should have no issues with them, e.g.
+
+[source,bash]
+----
+[root@75d1df96849c data]# objdump -t ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner | grep “GreetingResource”
+----
+
+Flame graphs should also be generated inside the container:
+
+[source,bash]
+----
+[root@75d1df96849c data]# perf script | /opt/FlameGraph/stackcollapse-perf.pl > out.perf-folded
+[root@75d1df96849c data]# /opt/FlameGraph/flamegraph.pl out.perf-folded > flamegraph.svg
+----
+
+The resulting svg files can then be open outside the container for visualization.
+
+=== Generating flame graphs is slow, or produces errors, what can I do?
+
+There are multiple ways in which a native binary produced by Mandrel can be profiled.
+All the methods require you to pass in the `-H:-DeleteLocalSymbols` option.
+
+The method shown in this reference guide generates a binary with DWARF debug information,
+runs it via `perf record` and then uses `perf script` and flame graph tooling to generate the flamegraphs.
+However, the `perf script` post-processing step done on this binary can appear to be slow or can show some DWARF errors.
+
+An alternative method to generate flame graphs is to pass in `-H:+PreserveFramePointer` when generating the native binary instead of generating the DWARF debug information.
+It instructs the binary to use an extra register for the frame pointer.
+This enables `perf` to do stack walking to profile the runtime behaviour.
+To generate the native binary using these flags, do the following:
+
+[source,bash,subs=attributes+]
+----
+./mvnw package -DskipTests -Pnative -Dquarkus.native.container-build=true \
+    -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:{mandrel-flavor} \
+    -Dquarkus.native.additional-build-args=-H:+PreserveFramePointer,-H:-DeleteLocalSymbols
+----
+
+To get runtime profiling information out of the native binary, simply do:
```

Comment:
General remark, we decided to use `native executable` in most of the other guides.